### PR TITLE
Add an option to disable 3rd party cookies in e2e tests

### PR DESCRIPTION
**Pull request by lsl**
#### Changes proposed in this Pull Request

* Adds an option to disable third party cookies in the browser settings used by e2e tests.
* nb: This change also adds previously omitted chrome browser settings in the saucelabs section (we can remove this, it just looks like an omission rather than something intentional) 

#### Testing instructions

I was testing this against #51618 but any spec will do.

I've added the commit 
f9adcf2 to disable 3pc for all tests to identify whether we can do this by default or if there are features that need fixing. (which would also be useful for us to know over at c3pop2.) 

If everything passes or the large majority of things do we should leave this disabled by default and enable them for the failing tests. If lots is failing we can remove the commit.

```
NODE_CONFIG_ENV=decrypted BROWSERSIZE=desktop ./node_modules/.bin/mocha specs/wp-likes-spec.js
```

Related to #51618

### CODE REVIEW

**Comment by WunderBart** [comment]
> If everything passes or the large majority of things do we should leave this disabled by default and enable them for the failing tests. If lots is failing we can remove the commit.

Looks like it's all good! Could you explain though why we want to disable 3pc in e2e tests?

**Comment by lsl** [comment]
[Chrome will be disabling third party cookies](https://blog.google/products/ads-commerce/a-more-privacy-first-web/) at some point, this test change should ensure our tests catch any failures that arise from this early.. though.. it seems we don't have any of these issues under test at the moment. (or there aren't any, or we've fixed them with the RLT work.)

This initiative is happening out of pcNnmV-6-p2 and there is a spreadsheet linked in pb6Nl-dlv-p2 which contains a list of known issues.

---

Before merging I want to:
- [x] Add a test we know will fail with 3pc disabled (and then skip it for the time being) - wip in #52145, will deploy working notifications test first, then skip it in this pr.
- [x] Drop the saucelabs change - not sure how to test this myself so I'd rather not break or leave anything broken. Its easy to add later if anyone needs it.
- [x] Check or drop the firefox test change - Firefox e2e tests are failing for another reason, going to leave this change as is.

**Comment by scinos** [comment]
Nice work!

FWIW, e2e are stuck in an old Chrome version because we found login problems when we updated it, likely caused by the cookie changes Chrome has been introducing lately. I wonder if this PR will unblock that update.

> Drop the saucelabs change

Afaik we don't use Saucelabs to test Chrome, I think it is safe to not change it.

**Review by scinos — APPROVED** [review]

**Comment by bsessions85** [comment]
I can't help but wonder if this will help us with using newer version of Chrome for e2e testing

**Comment by lsl** [comment]
@bsessions85 

> I can't help but wonder if this will help us with using newer version of Chrome for e2e testing

It shouldn't, but is there an issue using v90 currently? (locally seems to run ok for the handful of tests I'm tinkering with)

**Comment by lsl** [comment]
Found the magic words 34bd419, things should blow up now :smile: :boom:

**Comment by lsl** [comment]
Or not!

#52145 is now failing though so - PR success. And since nothing under test in calypso is failing due to 3pc we can deploy this as is without removing 25c7580 like I planned.

**Comment by lsl** [comment]
Correction, #52145 fails locally, passes on CI. Trying out an older setting option that might work for CI, [`ece683c` (#52145)](https://github.com/Automattic/wp-calypso/pull/52145/commits/ece683c8754b850bbdecb9e027e1969ff7610ed6), but didn't work locally on v90.

**Comment by lsl** [comment]
Well that caused tests to [fail spectacularly :+1: ](https://teamcity.a8c.com/viewLog.html?buildId=6130386&buildTypeId=calypso_RunCalypsoE2eDesktopTests). So I guess the differences between the old version 84 [specified here?](https://github.com/Automattic/wp-calypso/blob/eebd38b3897d314f12b5d37daddbe2ce74fc34e1/Dockerfile.base#L125) and my local browser were causing a settings mismatch to not apply 3pc blocking correctly.

---

@bsessions85 Was it insecure SameSite=None cookies what was holding us back on chrome versions? (its unclear from the git-blame)

@jsnajdr removed the insecure ones in D58337-code but they shouldn't have caused a problem for the browser anyway. 

Is confirming this as simple as bumping the chrome version number in that Dockerfile and running CI against the PR or does a merge need to happen before CI picks up the change?

**Comment by scinos** [comment]
> Is confirming this as simple as bumping the chrome version number in that Dockerfile and running CI against the PR or does a merge need to happen before CI picks up the change?

You'll need to update Chrome _in your branch_ first, by changing the TeamCity template and adding the required `apt-get` commands (I can help with that).

Once that is working and merged, we can "lift" that change to the Dockerfile so future runs will have the correct Chrome version already installed.

**Comment by lsl** [comment]
I want to lock the understanding of these settings into code so I've made the settings behave like chrome defaults - tests should pass now. 

Lets merge this with dual version support for now, I'll follow up with another PR for updating chrome and mess around with actually disabling 3pc cookies in either another PR or #52145 (if its even worth bothering to continue with)

**Pull request merged**
